### PR TITLE
Add check of Minimum supported Jenkins version on startup

### LINKED ISSUE #191: Jenkinsfile Runner should fail immediately when running on an unsupported Jenkins version
`jenkins.version` in pom.xml actually defines the minimum supported Jenkins version. But this version is not really checked anywhere, and JFR run may fail randomly. It would be nice to verify the Jenkins version before staring execution.

Jenkins WAR file includes the Jenkins core version in its properties: https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/jenkins/model/Jenkins.java#L5057-L5066 . The code can be copied from there

It is a follow-up to https://github.com/jenkinsci/jenkinsfile-runner/issues/170

### PR DESCRIPTION
**Pull request by glefloch**
closes #191 by adding a check on version number before doing anything.

### CODE REVIEW

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Thanks! AFAICT the comparison needs to be changed to "greater or equal", because we define minimum Jenkins version, not the only required one

**Comment by glefloch** [comment]
@oleg-nenashev yes, I wanted to do that, but, can I safely convert the version from `String` to `double` ?

**Comment by oleg-nenashev** [comment]
I suggest using https://github.com/jenkinsci/lib-version-number/blob/master/src/main/java/hudson/util/VersionNumber.java which is a standard way to compare versions in Jenkins

File: bootstrap/pom.xml [hunk 1]
```diff
@@ -18,5 +18,10 @@
       <groupId>commons-io</groupId>
       <artifactId>commons-io</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci</groupId>
+      <artifactId>version-number</artifactId>
+      <version>1.6</version>
```
Comment (oleg-nenashev) [inline bootstrap/pom.xml:24]: Will need to take it from BOM after #177

**Review by oleg-nenashev — APPROVED** [review]
Thanks!

**Pull request merged**